### PR TITLE
Create a review

### PR DESCRIPTION
**Pull request by giftnichakul**
- Add review
  image

### CODE REVIEW

**Review by kanatsanan6 — COMMENTED** [review]
Instead of separating the PR to several branch and merge them all into 1 branch. If some branches are continue from one another, you can also do something like below example. 

```
                        feature/add-book  feature/add-comment
   feature/show-book         |            |
          |                  |            |
          |------------------|------------|
main      |
  |       |
  |-------|
  |
```

File: app/controllers/reviews_controller.rb [hunk 1]
```diff
@@ -1,2 +1,13 @@
...
+  def create
+    @book = Book.find(params[:book_id])
+    @review = @book.reviews.create(review_params)
+    redirect_to book_path(@book)
+  end
+
+  private
+
+  def review_params
+    params.require(:review).permit(:comment, :star, :book)
```
Comment (WatcharitNew) [inline app/controllers/reviews_controller.rb]: optional:
```
params.require(:review).permit(:comment, :star, :book_id)
```

File: app/controllers/reviews_controller.rb [hunk 2]
```diff
@@ -1,2 +1,13 @@
 class ReviewsController < ApplicationController
+  def create
+    @book = Book.find(params[:book_id])
+    @review = @book.reviews.create(review_params)
```
Comment (WatcharitNew) [inline app/controllers/reviews_controller.rb:4]: If you do as above comments, you can skip find `Book`
and only create review

```
@review = Review.create(review_params)
```

File: app/controllers/reviews_controller.rb [hunk 3]
```diff
@@ -1,2 +1,13 @@
...
+    redirect_to book_path(@book)
```
Comment (WatcharitNew) [inline app/controllers/reviews_controller.rb:5]: For this, you can change to
```
redirect_to book_path(id: params[:book_id])
```

**Review by WatcharitNew — APPROVED** [review]

**Comment by giftnichakul** [comment]
> Instead of separating the PR to several branch and merge them all into 1 branch. If some branches are continue from one another, you can also do something like below example.
> 
> ```
>                         feature/add-book  feature/add-comment
>    feature/show-book         |            |
>           |                  |            |
>           |------------------|------------|
> main      |
>   |       |
>   |-------|
>   |
> ```

อาเครค่ะ🫡

File: app/controllers/reviews_controller.rb (hunk 3 above)
Comment (komsun-k) [reply app/controllers/reviews_controller.rb:5]: Agreed on this point.

File: app/controllers/reviews_controller.rb (hunk 2 above)
Comment (giftnichakul) [reply app/controllers/reviews_controller.rb:4]: พอลองแก้ตามนี้แล้วต้องแก้ review_params เป็นแบบนี้มันถึงจะได้ผลอะค่ะ พอรู้สาเหตุไหมอะคะ 🥹
```
def review_params
    params.require(:review).permit(:comment, :star).merge(book_id: params[:book_id])
    # params.require(:review).permit(:comment, :star, :book_id)
end
```

**Pull request merged**